api/internal/localizer/util.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"log"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// validateScope returns the scope given localize arguments and targetLdr that corresponds to targetArg
+func validateScope(scopeArg string, targetArg string, targetLdr ifc.Loader, fSys filesys.FileSystem) (filesys.ConfirmedDir, error) {
+	if _, isRemote := targetLdr.Repo(); isRemote {
+		if scopeArg != "" {
+			return "", errors.Errorf("scope '%s' specified for remote localize target '%s'", scopeArg, targetArg)
+		}
+		return "", nil
+	}
+	// default scope
+	if scopeArg == "" {
+		return filesys.ConfirmedDir(targetLdr.Root()), nil
+	}
+	scope, err := filesys.ConfirmDir(fSys, scopeArg)
+	if err != nil {
+		return "", errors.WrapPrefixf(err, "unable to establish localize scope")
+	}
+	if !filesys.ConfirmedDir(targetLdr.Root()).HasPrefix(scope) {
+		return scope, errors.Errorf("localize scope '%s' does not contain target '%s' at '%s'",
+			scopeArg, targetArg, targetLdr.Root())
+	}
+	return scope, nil
+}
+
+// validateNewDir returns the localize destination directory or error. Note that spec is nil if targetLdr is at local
+// target.
+func validateNewDir(newDirArg string, targetLdr ifc.Loader, spec *git.RepoSpec, fSys filesys.FileSystem) (filesys.ConfirmedDir, error) {
+	var newDirPath string
+	if newDirArg == "" {
+		newDirPath = defaultNewDir(targetLdr, spec)
+	} else {
+		newDirPath = newDirArg
+	}
```

Comment:
nit: A more concise pattern would be like

```
func validateNewDir(newDir string, targetLdr ifc.Loader, spec *git.RepoSpec, fSys filesys.FileSystem) (filesys.ConfirmedDir, error) {
	if newDir== "" {
		newDir = defaultNewDir(targetLdr, spec)
    }
```

There may be some places in your other functions as well; see if you can just use `newDir`, `target`, and `scope` instead of `newDirArg` etc and modify the variables where needed.

Reply:
I made the change in this case. I originally separated `newDirArg` and `newDirPath` into different variables because they represent different values: the first being the argument the user passed in and the second being the path that we try to create. I thought it'd be more readable for each variable to have a single purpose.

I didn't change the other instances you mentioned because the variables have different types. The variables named `xArg` are strings, whereas their `x` counterparts are `filesys.ConfirmedDir`.